### PR TITLE
Adding optional income/tax chart

### PR DESCRIPTION
**Pull request by AchoArnold**

### CODE REVIEW

File: src/calc/calc.component.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import template from './calc.html';
 import constants from '../data.json'; // Get JSON containing calculation constants
-
+import {CHART_NET_INCOME_NET_TAX} from './constants';
 
 let calcComponent = {
   template,
-  controller: function($scope, $location) {
+  controller: function($scope, $location, $element) {
```
Comment (stevermeister) [inline src/calc/calc.component.js]: what was the reason to inject `$element` here?

File: src/calc/calc.component.js [hunk 2]
```diff
@@ -329,8 +341,80 @@ let calcComponent = {
       });
       return amount;
     }
+
+    /**
+     * This function clears the chart element.
+     * This is a walk-around since we can't call the `chart.destroy()` function.
```
Comment (stevermeister) [inline src/calc/calc.component.js]: > we can't call the `chart.destroy()`

why?

File: src/index.html [hunk 3]
```diff
@@ -28,14 +28,16 @@
 		</md-container>
 		<ruling></ruling>
 		<partner-section></partner-section>
-		<world-calc-section></world-calc->
```
Comment (stevermeister) [inline src/index.html]: OMG

File: src/calc/calc.component.js [hunk 4]
```diff
@@ -329,8 +341,80 @@ let calcComponent = {
...
+     *
+     * @param {object} salary
+     */
+    function drawIncomeChart(salary) {
+      let chartColors = {
+        red: 'rgb(255, 99, 132)',
+        green: 'rgb(75, 192, 192)',
+      };
+
+      let chartElement = angular.element(document.getElementById(CHART_NET_INCOME_NET_TAX));
```
Comment (stevermeister) [inline src/calc/calc.component.js]: maybe it's more reasonable to use `angular-chart.js`?
https://github.com/jtblin/angular-chart.js

**Comment by stevermeister** [comment]
>  pull bot closed this 1 minute ago

frankly speaking, don't know what does it mean

File: src/calc/calc.component.js (hunk 1 above)
Comment (AchoArnold) [reply src/calc/calc.component.js]: Removed

File: src/calc/calc.component.js (hunk 2 above)
Comment (AchoArnold) [reply src/calc/calc.component.js]: Fixed 👍

File: src/calc/calc.component.js (hunk 4 above)
Comment (AchoArnold) [reply src/calc/calc.component.js]: Well I didn't want to add the extra dependency. `angular-chart` still requires you to use chart.js

**Pull request closed without merging**